math/igraph/Portfile

```diff
@@ -30,36 +31,41 @@ checksums           rmd160  433e8cb78287600b6d8b2d9173faa2e8ea5d4f42 \
 test.run            yes
 test.target         check
 
+compiler.c_standard 1999
 compiler.cxx_standard \
                     2011
 
-# Build options for igraph:
-#  - Do not use ccache to build this port unless MacPorts tells it to.
-#  - Build a shared library.
-#  - Enable link-time optimization when available.
-#  - Set features and the use of externaly libraries explicitly---do not leave this to auto-detection.
-#  - Use the vecLib BLAS/LAPACK.
-configure.args-append \
-                    -DUSE_CCACHE=OFF \
-                    -DBUILD_SHARED_LIBS=ON \
-                    -DIGRAPH_ENABLE_LTO=AUTO \
-                    -DIGRAPH_GLPK_SUPPORT=ON \
-                    -DIGRAPH_GRAPHML_SUPPORT=ON \
-                    -DIGRAPH_USE_INTERNAL_ARPACK=OFF \
-                    -DIGRAPH_USE_INTERNAL_BLAS=OFF \
-                    -DIGRAPH_USE_INTERNAL_CXSPARSE=OFF \
-                    -DIGRAPH_USE_INTERNAL_GLPK=OFF \
-                    -DIGRAPH_USE_INTERNAL_GMP=OFF \
-                    -DIGRAPH_USE_INTERNAL_LAPACK=OFF \
-                    -DBLA_VENDOR=Apple
+linalg.setup        noveclibfort
+
+pre-configure {
+    # Build options for igraph:
+    #  - Do not use ccache to build this port unless MacPorts tells it to.
+    #  - Build a shared library.
+    #  - Enable link-time optimization when available.
+    #  - Set features and the use of external libraries explicitly---do not leave this to auto-detection.
+    configure.args-append   -DUSE_CCACHE=OFF \
+                            -DBUILD_SHARED_LIBS=ON \
+                            -DIGRAPH_ENABLE_LTO=AUTO \
+                            -DIGRAPH_GLPK_SUPPORT=ON \
+                            -DIGRAPH_GRAPHML_SUPPORT=ON \
+                            -DIGRAPH_USE_INTERNAL_ARPACK=OFF \
+                            -DIGRAPH_USE_INTERNAL_BLAS=OFF \
+                            -DIGRAPH_USE_INTERNAL_CXSPARSE=OFF \
+                            -DIGRAPH_USE_INTERNAL_GLPK=OFF \
+                            -DIGRAPH_USE_INTERNAL_GMP=OFF \
+                            -DIGRAPH_USE_INTERNAL_LAPACK=OFF
+
+    # Link to chosen BLAS/LAPACK
+    configure.args-append   ${cmake_linalglib}
+}
+
+variant tls description {Build thread-safe version} {
+    compiler.thread_local_storage   yes
+    configure.args-append           -DIGRAPH_ENABLE_TLS=ON
+}
```

Comment:
Why is this a variant? Why not always enable this? No dependencies are added.

Reply:
@mf2k I would like to have a consistent build on all versions of macOS, with the same features. I don't know if it will build correctly on older macOS with TLS. I think it won't on 10.6.

I believe most users of igraph don't need TLS. Also, thread-safety is not mature or well tested in igraph. Finally, it depends on the thread safety of dependencies as well. E.g. GLPK is only built to be thread safe on macOS 10.11 and later.  I kept it off by default for these reasons.

Would it be possible to keep tls as a variant, and have it enabled by default only for recent macOS versions? Is this possible with MacPorts? I know it is possible to compile with TLS only on some versions, but having an explicit variant, as it would make it absolutely clear to users what features are enabled in the installed build.

Reply:
I force-pushed a version which enables it unconditionally, but I would prefer not to go with that option (and break the port on old macOS) if possible.

Reply:
I see. You are the maintainer. If you need the variant to support older versions of macOs, that is fine.

Reply:
Make the variant default though because most people are on macOS beyond 10.6.

Reply:
Actually you can make it conditionally default for macOS > 10.6.

Reply:
Done, please have another look Frank.